src/components/accordion/accordion.test.js

```diff
@@ -0,0 +1,539 @@
+          // set by default
+          accordionTitleContainer()
+            .should("have.css", "justify-content", "space-between")
+            .and("not.have.css", "flex-direction", "row-reverse");
+          accordionTitleContainer(positionOfElement("first"))
+            .first()
+            .should("have.css", "margin-right", "0px");
+        } else {
+          accordionTitleContainer().should(
+            "have.css",
+            "flex-direction",
+            "row-reverse"
+          );
+          accordionTitleContainerByPosition(positionOfElement("first"))
+            .last()
+            .should("have.css", "margin-right", "16px");
+        }
+      }
+    );
+
+    it("should verify AccordionRow is expanded by clicking on validation icon", () => {
+      CypressMountWithProviders(<AccordionWithIcon />);
+
+      accordionIcon().eq(0).click();
+
+      accordionTitleContainer()
+        .should("have.attr", "aria-expanded", "true")
+        .and("be.visible");
+
+      accordionContent()
+        .should("have.attr", "data-element", "accordion-content")
+        .and("be.visible");
+    });
+
+    it.each([[true], [false]])(
+      "should check onChange action was called when expanded is set to %s",
+      (isExpanded) => {
+        const callback = cy.stub();
+
+        CypressMountWithProviders(
+          <AccordionComponent expanded={isExpanded} onChange={callback} />
+        );
+
+        accordionTitleContainer()
+          .click()
+          .then(() => {
+            // eslint-disable-next-line no-unused-expressions
+            expect(callback).to.have.been.calledOnce;
+          });
+      }
+    );
+
+    it.each(specialCharacters)(
+      "should render Accordion component with %s as a title",
+      (titleValue) => {
+        CypressMountWithProviders(<AccordionComponent title={titleValue} />);
+
+        accordionTitleContainer().should("contain.text", titleValue);
+      }
+    );
+
+    it.each(specialCharacters)(
+      "should render Accordion component with %s as a subtitle",
+      (titleValue) => {
+        CypressMountWithProviders(<AccordionComponent subTitle={titleValue} />);
+
+        accordionTitleContainer().should("contain.text", titleValue);
+      }
+    );
+
+    it.each(sizes)(
+      "should render Accordion component with %s as a size and has height property set to %s",
+      (size, height) => {
+        CypressMountWithProviders(<AccordionComponent size={size} />);
+
+        accordionTitleContainer()
+          .should("have.css", "height")
+          .and("contain", height);
+      }
+    );
+
+    it.each([["full"], ["none"]])(
+      "should render Accordion component with border type '%s'",
+      (borders) => {
+        CypressMountWithProviders(<AccordionComponent borders={borders} />);
+
+        if (borders === "none") {
+          accordion()
+            .should("have.css", "border-bottom-style", "none")
+            .and("have.css", "border-bottom-color", "rgba(0, 0, 0, 0.9)");
```

Comment:
we can pass a second parameter in `(borders, color)` and use it here avoiding using `if` statement as:
```suggestion
            .and("have.css", "border-bottom-color", color);
```